classes/class-db-driver-wpdb.php

```diff
@@ -173,6 +173,20 @@ public function setup_storage( $plugin ) {
 	 * @return \WP_Stream\Uninstall
 	 */
 	public function purge_storage( $plugin ) {
+		// Avoid duplicate uninstall action.
+		// TODO: This is nasty code right now due to the fact the object instance is only known to the action.
+		// The architecture around how the uninstall action is registered should be rethought.
+		if ( has_action( 'wp_ajax_wp_stream_uninstall' ) ) {
+			// Get a list of all callbacks registered to the default priority 10 for the uninstall action.
+			$callbacks = array_values( $GLOBALS['wp_filter']['wp_ajax_wp_stream_uninstall']->callbacks[10] );
+			// From that list, retrieve function of the first (and only) callback.
+			$callback_function = $callbacks[0]['function'];
+			// The callback's first element should be the object instance of the Uninstall class.
+			$uninstall = $callback_function[0];
+
+			return $uninstall;
+		}
+
```

Comment:
@kasparsd During testing, the `purge_storage()` method is sometimes triggered twice. Any suggestions on how to deal with this in a reliable without rewriting parts of the architecture?
Do you think prefixing this hack with a check on `WP_STREAM_TESTS` would be enough for now?

Reply:
This was happening because every test calling `purge_storage()` was creating a new instance of `Install()` which was added as new `wp_ajax_wp_stream_uninstall` action callback which then all got called when we triggered the action.

```php
add_action( 'aaa', [ new Uninstall( $this->plugin ), 'uninstall' ] );
add_action( 'aaa', [ new Uninstall( $this->plugin ), 'uninstall' ] );
```

results in:

```php
/var/www/html/wp-content/plugins/stream-src/tests/tests/test-class-db-driver-wpdb.php:33:
class WP_Hook#43 (5) {
  public $callbacks =>
  array(1) {
    [10] =>
    array(2) {
      '000000000a8a85d000000000584fc7efuninstall' =>
      array(2) {
        ...
      }
      '000000000a8a85d600000000584fc7efuninstall' =>
      array(2) {
        ...
      }
    }
  }
```
```